src/Components/Blazor/WebAssembly.Authentication/src/AuthenticationManager.cs

```diff
@@ -0,0 +1,446 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components.Rendering;
+using Microsoft.JSInterop;
+
+namespace Microsoft.AspNetCore.Components.WebAssembly.Authentication
+{
+    /// <summary>
+    /// A component that handles remote authentication operations in an application.
+    /// </summary>
+    /// <typeparam name="TAuthenticationState">The user state type persisted while the operation is in progress. It must be serializable.</typeparam>
+    public class AuthenticationManager<TAuthenticationState> : ComponentBase where TAuthenticationState : RemoteAuthenticationState
```

Comment:
This is a very general-sounding name, so there's a risk that our naming will be hard for people to understand. We will have an authentication "service", an authentication "manager", and an authentication "provider". TBH I'd struggle to remember which one was for which purpose!

To mitigate this, could we rename `AuthenticationManager<T>` to something that really emphasises it's for the "remote" authentication case, and is about providing a UI over an underlying `RemoteAuthenticationService`?

Suggestion: `RemoteAuthenticationServiceView` or `RemoteAuthenticationServiceDisplay`